src/computational_graph/diagram.jl

```diff
@@ -169,12 +222,62 @@ function Base.:+(g1::Diagram{F,W}, g2::Diagram{F,W}) where {F,W}
     type = g1.type
     @assert Set(g1.external_vertices) == Set(g2.external_vertices)
     #TODO: add external vertices creation/annihilation check
-    return Diagram{F,W}(g1.external_vertices; type=type, is_connected=g1.is_connected && g2.is_connected, subdiagram=[g1, g2], operator=Sum())
+    return Diagram{F,W}(g1.external_vertices; type=type, is_connected=g1.is_connected && g2.is_connected, couplings=g1.couplings, subdiagram=[g1, g2], operator=Sum())
 end
 
 function Base.:-(g1::Diagram{F,W}, g2::Diagram{F,W}) where {F,W}
     return g1 + (-1) * g2
 end
+
+"""
+    function ExternalVertices(V1::Vector{ExternalVertex}, V2::Vector{ExternalVertex}, couplings::Couplings)
+
+    Give external vertices when two diagrams combine.
+    
+    # Arguments
+    - V1::Vector{ExternalVertex}     : External vertices of diagram I.
+    - V2::Vector{ExternalVertex}     : External vertices of diagram II.
+    - couplings::Couplings           : all the vertex couplings in the Feynman rule. 
+"""
+function ExternalVertices(V1::Vector{ExternalVertex}, V2::Vector{ExternalVertex}, couplings::Couplings)
+    V1_ind = [v.point for v in V1]
+    V2_ind = [v.point for v in V2]
+    common = intersect(V1_ind, V2_ind)
+    total = union(V1, V2)
+    extV = [v for v in total if v.point ∉ common]
+    for point in common
+        ifinternal, ifexternal = false, false
+        i1 = findfirst(isequal(point), V1_ind)
+        i2 = findfirst(isequal(point), V2_ind)
+        point_ops = [V1[i1].operators..., V2[i2].operators...]
```

Comment:
This line can be rewritten as `point_ops = [V1[i1].operators; V2[i2].operators]` (see [the style guide](https://docs.julialang.org/en/v1/manual/style-guide/#Don't-overuse-...))